integration-test-groups/aws2/aws2-sqs-sns/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsSnsTest.java

```diff
@@ -40,28 +45,157 @@
 
     @Test
     public void sqs() {
-        final Config config = ConfigProvider.getConfig();
-        final String queueName = config.getValue("aws-sqs.queue-name", String.class);
+        final String queueName = getPredefinedQueueName();
 
-        String[] queues = RestAssured.get("/aws2-sqs-sns/sqs/queues")
+        final String[] queues = listQueues();
+        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+
+        final String msg = sendSingleMessageToQueue(queueName);
+        awaitMessageFromQueue(msg, queueName);
+    }
+
+    private String[] listQueues() {
+        return RestAssured.get("/aws2-sqs-sns/sqs/queues")
                 .then()
                 .statusCode(200)
                 .extract()
                 .body().as(String[].class);
-        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+    }
+
+    @Test
+    public void sqsDeleteMessage() {
+        final String qName = getPredefinedQueueName();
+        final String msg = sendSingleMessageToQueue(qName);
+        final String receipt = receiveReceiptOfMessageFromQueue(qName);
+        deleteMessageFromQueue(qName, receipt);
+        Assertions.assertNotEquals(receiveMessageFromQueue(qName), msg);
+    }
+
+    private String getPredefinedQueueName() {
+        return ConfigProvider.getConfig().getValue("aws-sqs.queue-name", String.class);
+    }
 
+    private String sendSingleMessageToQueue(String queueName) {
         final String msg = "sqs" + UUID.randomUUID().toString().replace("-", "");
         RestAssured.given()
                 .contentType(ContentType.TEXT)
                 .body(msg)
-                .post("/aws2-sqs-sns/sqs/send")
+                .post("/aws2-sqs-sns/sqs/send/" + queueName)
                 .then()
                 .statusCode(201);
+        return msg;
+    }
+
+    private String receiveReceiptOfMessageFromQueue(String queueName) {
+        return RestAssured.get("/aws2-sqs-sns/sqs/receive/receipt/" + queueName)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .asString();
+    }
+
+    private void deleteMessageFromQueue(String queueName, String receipt) {
+        RestAssured.delete("/aws2-sqs-sns/sqs/delete/message/" + queueName + "/" + receipt)
+                .then()
+                .statusCode(200);
+    }
+
+    @Test
+    public void sqsAutoCreateDelayedQueue() {
+        final String qName = "delayQueue";
+        final int delay = 10;
+        createDelayQueueAndVerifyExistence("delayQueue", 10);
```

Comment:
Would it be right to write `createDelayQueueAndVerifyExistence("delayQueue", delay);` ?

Reply:
definitely. Sorry, this method was refactored multiple times before pushed. I'll fix it right away